lib/chef/cookbook_version.rb

```diff
@@ -96,7 +96,13 @@ def root_dir
     # This is the one and only method that knows how cookbook files'
     # checksums are generated.
     def self.checksum_cookbook_file(filepath)
-      Chef::Digester.generate_md5_checksum_for_file(filepath)
+      if Chef::Config.fips_mode
+        # This will require a chef server that can handle
+        # sha256 checksums
```

Comment:
This is a large undertaking, but would be a win for everyone. If we have enough customer demand that we're gonna do this, perhaps we could use Chef Server API versioning to make the transition easier. Also, does FIPS mode completely disable MD5? For a smooth upgrade, you'd want to probably include *both* SHA2-256 and MD5, so that a newer version of knife could upload a cookbook and Chef Server would have all the info required to serve it to older chef-client.